File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number){
  int sum = 0;
  do
    sum += number*number;
  while (number--);

  return sum;
}

unsigned int square_of_sum(unsigned int number){
  int sum = 0;
  do
    sum += number;
  while (number--);

  return sum*sum;
}

unsigned int difference_of_squares(unsigned int number){
  return square_of_sum(number) - sum_of_squares(number);
}

File: javascript/difference-of-squares/difference-of-squares.js
// found here: https://learnersbucket.com/examples/algorithms/difference-between-square-of-sum-of-numbers-and-sum-of-square-of-numbers/#google_vignette

export class Squares {
  #num
  constructor(number) {
    this.#num = number;
  }

  get sumOfSquares() {
    return (this.#num * (this.#num + 1) * ((this.#num * 2) + 1)) / 6
  }

  get squareOfSum() {
    return ((this.#num * (this.#num + 1)) / 2) ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
